src/iso22133object.cpp

```diff
@@ -170,62 +145,43 @@ void TestObject::sendGREM(HeaderType msgHeader, GeneralResponseStatus responseCo
 }
 
 void TestObject::sendMonrLoop() {
-	std::stringstream ss;
-	ss << "Started MONR thread." << std::endl;
-	std::cout << ss.str();
-
-	while (this->on && ctrlChannel.isOpen()) {
-		// Only send monr if transmitterID has been set by an OSEM message
-		if (this->transmitterID != TRANSMITTER_ID_UNAVAILABLE_VALUE){
-			sendMONR();
+	while (this->on) {
+		if (ctrlChannel.isOpen()) {
+			// Only send monr if transmitterID has been set by an OSEM message
+			if (this->transmitterID != TRANSMITTER_ID_UNAVAILABLE_VALUE){
+				sendMONR();
+			}
+			auto t = std::chrono::steady_clock::now();
+			std::this_thread::sleep_until(t + monrPeriod);
 		}
-		auto t = std::chrono::steady_clock::now();
-		std::this_thread::sleep_until(t + monrPeriod);
 	}
-
-	ss.str(std::string());
-	ss << "Exiting MONR thread." << std::endl;
-	std::cout << ss.str();
 }
 
 void TestObject::receiveUDP() {
 	std::stringstream ss;
-	ss << "Started UDP communication thread." << std::endl;
-	ss << "Awaiting UDP data from ATOS..." << std::endl;
-	std::cout << ss.str();
-
 	awaitingFirstHeab = true;
 
-	while (this->on && ctrlChannel.isOpen()) {
-		auto data = processChannel.receive();
-
-		// Connection lost
-		if (data.size() <= 0) {
-			continue;
-		}
+	while (this->on) {
+		if (ctrlChannel.isOpen()) {
+			auto data = processChannel.receive();
 
-		if (awaitingFirstHeab) {
-			ss.str(std::string());
-			ss << "Received UDP data from ATOS" << std::endl;
-			std::cout << ss.str();
+			// Connection lost
+			if (data.size() <= 0) {
+				continue;
+			}
 
-			startSendMonr(); // UDP connection established, start sending MONR
+			int nBytesHandled = 0;
+			do {
+				try {
+					nBytesHandled = handleMessage(data);
+				} catch (const std::exception& e) {
+					std::cerr << e.what() << std::endl;
+					break;
+				}
+				data.erase(data.begin(), data.begin() + nBytesHandled);
+			} while (data.size() > 0);
 		}
-
-		int nBytesHandled = 0;
-		do {
-			try {
-				nBytesHandled = handleMessage(data);
-			} catch (const std::exception& e) {
-				std::cerr << e.what() << std::endl;
-				break;
-			}
-			data.erase(data.begin(), data.begin() + nBytesHandled);
-		} while (data.size() > 0);
 	}
-	ss.str(std::string());
-	ss << "Exiting UDP communication thread." << std::endl;
-	std::cout << ss.str();
```

Comment:
Printing is useful, but good to remove printing to stdout. We need a more sustainable logging solution.